src/main/java/pl/golabm/service/FamilyService.java

```diff
@@ -0,0 +1,43 @@
+package pl.golabm.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import pl.golabm.model.Family;
+import pl.golabm.repository.FamilyRepository;
+
+import java.util.Optional;
+
+@Service
+public class FamilyService {
+
+    private final FamilyRepository familyRepository;
+
+    @Autowired
+    FamilyService(final FamilyRepository familyRepository) {
+        this.familyRepository = familyRepository;
+    }
+
+    public Family save(Family family) {
+        return familyRepository.save(family);
+    }
+
+    public void delete(Long id) {
+        familyRepository.deleteById(id);
+    }
+
+    public Iterable<Family> getAll() {
+        return familyRepository.findAll();
+    }
+
+    public Family update(Long id, Family family) {
+        Optional<Family> oldFammily = familyRepository.findById(id);
```

Comment:
Set id to family property and use familyRepository.save(family). This should automatically update all props.

Reply:
Could you explain that more? How to set id to family property?

Reply:
family.setid(id);
return familyRepository.save(family);

:)